cypress-io/cypress
# fix: properly run multiple specs in run and headed mode on linux and windows in chrome

- Closes #22155 

### User facing changelog


Users can now properly `cypress run` multiple specs in headed mode on linux and windows in chrome

### Additional details


In 10.0.0 we updated run mode to close a browser tab and open a new one in between specs rather than killing the browser instance and launching a new one. Headed mode on linux and windows works slightly differently than in other circumstances in that when the last browser tab closes, the browser process exits. This prevents us from being able to launch a new tab since we no longer have a connection to the browser. We will fix this by launching a new tab to `about:blank` prior to closing the last tab. This will allow us to maintain that connection.

### Steps to test


1. Pick a project with multiple e2e specs
2. On linux and/or windows, run `cypress run --headed --browser --chrome` for the project
3. Ensure that all specs succeed

### How has the user experience changed?


Before:

https://user-images.githubusercontent.com/4873279/172425241-3871575d-fc6e-49d6-8f94-418b222bcd11.mov

After:

https://user-images.githubusercontent.com/4873279/172425210-16467a9a-da2b-470f-95e7-819dda980b00.mov

### PR Tasks


- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
